File: webapi/src/Core/BovIQ.Domain/Entities/Cow.cs
namespace BovIQ.Domain.Entities;

public class Cow
{
    public int Id { get; set; }
    public int BreedId { get; set; }
    public Breed Breed { get; set; } = default!;
    //public int HerdId { get; set; }
    //public Herd Herd { get; set; } = default!;
    public string? Name { get; set; } = string.Empty;
    public string EarTag { get; set; } = string.Empty;
    public DateTime FirstCalvingDate { get; set; }
    public DateTime DateOfBirth { get; set; }
    public ICollection<MilkSession> MilkSessions { get; set; } = [];
}

File: client-boviq/src/app/core/models/cows/cow.ts
import { Breed } from "../breeds/breed";

export interface Cattle {
  id: number;
  breedI: string;
  name: string;
  earTag: string;
  Breed: Breed;
}

export interface CreateCowRequest {
  breedI: string;
  name: string;
  earTag: string;
  firstCalvingDate: string;
  dateOfBirth: string;
}

export interface CreateCowRequest {
  breedI: string;
  name: string;
  earTag: string;
  firstCalvingDate: string;
  dateOfBirth: string;
}
